src/main/java/com/training/web/pages/flipkart/AppPage.java

```diff
@@ -0,0 +1,39 @@
+package com.training.web.pages.flipkart;
+
+import java.util.Iterator;
+import java.util.Set;
+
+import static com.codeborne.selenide.Selenide.$x;
+import static com.codeborne.selenide.WebDriverRunner.getWebDriver;
+
+public class AppPage {
+    private String parentWindow=getWebDriver().getWindowHandle();
```

Comment:
why is this used globally?

Reply:
this is used in 2 methods

Reply:
Which 2 methods? - I only see this in this method - `getUrl()`